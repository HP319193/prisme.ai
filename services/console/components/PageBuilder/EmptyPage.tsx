import { Trans, useTranslation } from 'next-i18next';
import plus from '../../icons/plus.svg';

interface SuggestionButtonProps {
  title: string;
  text: string;
  className?: string;
  color?: string;
  onClick?: () => void;
}

const SuggestionButton = ({
  title,
  text,
  className,
  color,
  onClick,
}: SuggestionButtonProps) => {
  return (
    <button
      className={`
        focus:outline-none
        flex flex-col rounded flex-1 m-2 p-4 text-sm text-left justify-between
        transition-all group
        hover:shadow-2xl hover:-translate-y-2
        ${className}
      `}
      style={{
        backgroundColor: color,
      }}
      onClick={onClick}
    >
      <div className="flex flex-col">
        <div className="font-bold mb-2">{title}</div>
        <div className="relative min-h-[55px]">
          <div className="transition-all opacity-100 group-hover:opacity-0">
            {text}
          </div>
          <div
            className="
            absolute top-0
        flex flex-row mt-4 py-2 px-4 rounded text-3xl
        bg-[white] transition-all whitespace-nowrap overflow-hidden
          opacity-0 group-hover:opacity-100
          "
            style={{ color }}
          >
            +
          </div>
        </div>
      </div>
    </button>
  );
};

interface EmptyPageProps {
  onAddBlock?: (blockName?: string) => void;
}

export const EmptyPage = ({ onAddBlock }: EmptyPageProps) => {
  const { t } = useTranslation('workspaces');
  return (
    <div className="flex flex-1 justify-between flex-col p-28">
      <div className="flex flex-col flex-1 justify-center w-[15rem]">
        <Trans t={t} i18nKey="pages.empty.title">
          <div className="font-bold text-2xl my-2">a</div>
          <div>b</div>
        </Trans>
        <div>
          <button
            className="flex flex-row bg-accent mt-4 p-4 rounded transition-all whitespace-nowrap overflow-hidden"
            onClick={() => onAddBlock && onAddBlock()}
          >
            {/* eslint-disable-next-line @next/next/no-img-element */}
            <img src={plus.src} alt="" className="w-[15px] h-[15px] m-1" />
          </button>
        </div>
      </div>
      <div className="flex flex-col">
        <div className="my-8">{t('pages.empty.subtitle')}</div>
        <div className="flex flex-row">
          <SuggestionButton
            title={t('pages.empty.suggestions.header.title')}
            text={t('pages.empty.suggestions.header.text')}
            className="!ml-0"
            color="#F7F8D0"
            onClick={() => onAddBlock && onAddBlock('Header')}
          />
          <SuggestionButton
            title={t('pages.empty.suggestions.cards.title')}
            text={t('pages.empty.suggestions.cards.text')}
            color="#FF9261"
            onClick={() => onAddBlock && onAddBlock('Cards')}
          />
          <SuggestionButton
            title={t('pages.empty.suggestions.richtext.title')}
            text={t('pages.empty.suggestions.richtext.text')}
            color="#E7F6F6"
            onClick={() => onAddBlock && onAddBlock('RichText')}
          />
          <SuggestionButton
            title={t('pages.empty.suggestions.form.title')}
            text={t('pages.empty.suggestions.form.text')}
            className="!mr-0"
            color="#F1ECFF"
            onClick={() => onAddBlock && onAddBlock('Form')}
          />
        </div>
      </div>
    </div>
  );
};

export default EmptyPage;
