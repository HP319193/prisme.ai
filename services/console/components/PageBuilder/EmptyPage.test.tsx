import renderer from 'react-test-renderer';
import EmptyPage from './EmptyPage';

it('should render', () => {
  const onAddBlock = jest.fn();
  const root = renderer.create(<EmptyPage onAddBlock={onAddBlock} />);
  expect(root.toJSON()).toMatchSnapshot();
  const els = root.root.findAllByType('button');

  els[0].props.onClick();
  expect(onAddBlock).toHaveBeenCalledWith();
  onAddBlock.mockClear();

  els[1].props.onClick();
  expect(onAddBlock).toHaveBeenCalledWith('Header');
  onAddBlock.mockClear();

  els[2].props.onClick();
  expect(onAddBlock).toHaveBeenCalledWith('Cards');
  onAddBlock.mockClear();

  els[3].props.onClick();
  expect(onAddBlock).toHaveBeenCalledWith('RichText');
  onAddBlock.mockClear();

  els[4].props.onClick();
  expect(onAddBlock).toHaveBeenCalledWith('Form');
  onAddBlock.mockClear();
});
