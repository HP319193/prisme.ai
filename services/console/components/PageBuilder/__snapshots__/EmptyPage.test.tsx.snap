// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="flex flex-1 justify-between flex-col p-28"
>
  <div
    className="flex flex-col flex-1 justify-center w-[15rem]"
  >
    <div
      className="font-bold text-2xl my-2"
    >
      a
    </div>
    <div>
      b
    </div>
    <div>
      <button
        className="flex flex-row bg-accent mt-4 p-4 rounded transition-all whitespace-nowrap overflow-hidden"
        onClick={[Function]}
      >
        <img
          alt=""
          className="w-[15px] h-[15px] m-1"
          src="/file.svg"
        />
      </button>
    </div>
  </div>
  <div
    className="flex flex-col"
  >
    <div
      className="my-8"
    >
      pages.empty.subtitle
    </div>
    <div
      className="flex flex-row"
    >
      <button
        className="
        focus:outline-none
        flex flex-col rounded flex-1 m-2 p-4 text-sm text-left justify-between
        transition-all group
        hover:shadow-2xl hover:-translate-y-2
        !ml-0
      "
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#F7F8D0",
          }
        }
      >
        <div
          className="flex flex-col"
        >
          <div
            className="font-bold mb-2"
          >
            pages.empty.suggestions.header.title
          </div>
          <div
            className="relative min-h-[55px]"
          >
            <div
              className="transition-all opacity-100 group-hover:opacity-0"
            >
              pages.empty.suggestions.header.text
            </div>
            <div
              className=" absolute top-0 flex flex-row mt-4 py-2 px-4 rounded text-3xl bg-[white] transition-all whitespace-nowrap overflow-hidden opacity-0 group-hover:opacity-100 "
              style={
                Object {
                  "color": "#F7F8D0",
                }
              }
            >
              +
            </div>
          </div>
        </div>
      </button>
      <button
        className="
        focus:outline-none
        flex flex-col rounded flex-1 m-2 p-4 text-sm text-left justify-between
        transition-all group
        hover:shadow-2xl hover:-translate-y-2
        undefined
      "
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#FF9261",
          }
        }
      >
        <div
          className="flex flex-col"
        >
          <div
            className="font-bold mb-2"
          >
            pages.empty.suggestions.cards.title
          </div>
          <div
            className="relative min-h-[55px]"
          >
            <div
              className="transition-all opacity-100 group-hover:opacity-0"
            >
              pages.empty.suggestions.cards.text
            </div>
            <div
              className=" absolute top-0 flex flex-row mt-4 py-2 px-4 rounded text-3xl bg-[white] transition-all whitespace-nowrap overflow-hidden opacity-0 group-hover:opacity-100 "
              style={
                Object {
                  "color": "#FF9261",
                }
              }
            >
              +
            </div>
          </div>
        </div>
      </button>
      <button
        className="
        focus:outline-none
        flex flex-col rounded flex-1 m-2 p-4 text-sm text-left justify-between
        transition-all group
        hover:shadow-2xl hover:-translate-y-2
        undefined
      "
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#E7F6F6",
          }
        }
      >
        <div
          className="flex flex-col"
        >
          <div
            className="font-bold mb-2"
          >
            pages.empty.suggestions.richtext.title
          </div>
          <div
            className="relative min-h-[55px]"
          >
            <div
              className="transition-all opacity-100 group-hover:opacity-0"
            >
              pages.empty.suggestions.richtext.text
            </div>
            <div
              className=" absolute top-0 flex flex-row mt-4 py-2 px-4 rounded text-3xl bg-[white] transition-all whitespace-nowrap overflow-hidden opacity-0 group-hover:opacity-100 "
              style={
                Object {
                  "color": "#E7F6F6",
                }
              }
            >
              +
            </div>
          </div>
        </div>
      </button>
      <button
        className="
        focus:outline-none
        flex flex-col rounded flex-1 m-2 p-4 text-sm text-left justify-between
        transition-all group
        hover:shadow-2xl hover:-translate-y-2
        !mr-0
      "
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#F1ECFF",
          }
        }
      >
        <div
          className="flex flex-col"
        >
          <div
            className="font-bold mb-2"
          >
            pages.empty.suggestions.form.title
          </div>
          <div
            className="relative min-h-[55px]"
          >
            <div
              className="transition-all opacity-100 group-hover:opacity-0"
            >
              pages.empty.suggestions.form.text
            </div>
            <div
              className=" absolute top-0 flex flex-row mt-4 py-2 px-4 rounded text-3xl bg-[white] transition-all whitespace-nowrap overflow-hidden opacity-0 group-hover:opacity-100 "
              style={
                Object {
                  "color": "#F1ECFF",
                }
              }
            >
              +
            </div>
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;
